app/frontend/util/ChampaignAPI.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import _ from 'lodash';
+import $ from 'jquery';
+import { camelizeKeys } from './util';
+
+export type OperationResponse = {
+  success: boolean,
+  errors: ?{[id:string]: string[]}
+}
+
+const parseResponse = (response):OperationResponse => {
+  if(response === undefined) {
+    return { success: true, errors: {} };
+  }
+
+  switch (response.status) {
+    case 200:
+    case 201:
+    case 204:
+      return { success: true, errors: {} };
+    case 422:
+      return { success: false, errors: camelizeKeys(response.responseJSON.errors) };
+    default:
+      return {
+        success: false,
+        errors: {
+          base: [<FormattedMessage id="call_tool.errors.unknown" defaultMessage={`Unknown error, code ${response.code}`} />]
+        }
+      };
+  }
+};
+
+
+const createCall = function(params: {pageId: string, memberPhoneNumber: string, targetIndex: string}) {
```

Comment:
Let's talk about how we want to organize the ChampaignAPI methods, this is just an initial stab at it, and my idea is to adapt it as it grows.

On fetching individual functions, I'm all in to do that if it's a strong convention in the js world. But otherwise I believe that `ChampaignAPI.createCall(....)` is way clearer about the intentions of the function than the plain `createCall(...)`.